### PR TITLE
[TechDocs] Expandable Navigation addon

### PR DESCRIPTION
**Pull request by realandersn**
## Hey, I just made a Pull Request!

A TechDocs addon making it possible to expand/collapse the main navigation and keeping your preference in local storage.

ExpandNav

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by camilaibs — COMMENTED** [review]
Super helpful, thanks for the contribution!
I just added a few suggestions, hope it makes sense to you 🙂

**Review by camilaibs — APPROVED** [review]

**Comment by camilaibs** [comment]
@realandersn pull request approved 🎉 
It would be nice to add this new addon to the list of available add-ons here:
https://backstage.io/docs/features/techdocs/addons#available-addons

File: packages/app/src/App.tsx [hunk 1]
```diff
@@ -182,6 +185,7 @@ const routes = (
     >
       {techDocsPage}
       <TechDocsAddons>
+        <ExpandableNavigation />
```
Comment (camilaibs) [inline packages/app/src/App.tsx:188]: Should I add it to the embedded app in the TechDocs CLI?

**Review by Rugvip — APPROVED** [review]
Nice :grin:

Just a tiny optional nit

**Pull request merged**